tests/rados/test_osd_ecpool_inconsistency_scenario.py

```diff
@@ -0,0 +1,285 @@
+            return 1
+        pg_id = rados_obj.create_ecpool_inconsistent_obj(
+            objectstore_obj, client_node, pool_name, no_of_objects
+        )
+        obj_count = get_inconsistent_count(scrub_object, pg_id, rados_obj, "deep-scrub")
+        if obj_count < 0:
+            log.error(f"The deep-scrub not initiated on the pg-{pg_id}")
+            return 1
+
+        if obj_count != 0:
+            log.error(
+                f"deep-scrub repaired the {no_of_objects - obj_count} inconsistent objects out of {no_of_objects} "
+                f"but the osd_scrub_auto_repair_num_errors count is 5"
+            )
+            return 1
+        # Checking the logs
+        log_line = [
+            "active+clean+scrubbing+deep+repair",
+        ]
+        if not verify_mgr_log(
+            rados_obj=rados_obj,
+            mgr_daemon=mgr_daemon,
+            start_time=init_time,
+            end_time=end_time,
+            lines=log_line,
+        ):
+            log.error(f"The message {log_line} not found in the mgr logs")
+            return 1
+    except Exception as e:
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Execution of finally block")
+        if config.get("delete_pool"):
+            method_should_succeed(rados_obj.delete_pool, pool_name)
+            log.info("deleted the pool successfully")
+        set_default_param_value(mon_obj)
+        # log cluster health
+        rados_obj.log_cluster_health()
+    return 0
+
+
+def get_inconsistent_count(scrub_object, pg_id, rados_obj, operation):
+    """
+    Perform scrub and get the inconsistent object count
+    Args:
+        pg_id: pg id
+        rados_obj: Rados object
+        operation: scrub/deep-scrub operation
+
+    Returns: No of inconsistent object count or negative value(-1) if scrub/deep-scrub not performed on PG
+
+    """
+    operation_chk_flag = False
+    osd_scrub_min_interval = 120
+    osd_scrub_max_interval = 3600
+    (
+        scrub_begin_hour,
+        scrub_begin_weekday,
+        scrub_end_hour,
+        scrub_end_weekday,
+    ) = scrub_object.add_begin_end_hours(0, 1)
+
+    scrub_object.set_osd_configuration("osd_scrub_begin_hour", scrub_begin_hour)
+    scrub_object.set_osd_configuration("osd_scrub_begin_week_day", scrub_begin_weekday)
+    scrub_object.set_osd_configuration("osd_scrub_end_hour", scrub_end_hour)
+    scrub_object.set_osd_configuration("osd_scrub_end_week_day", scrub_end_weekday)
+    scrub_object.set_osd_configuration("osd_scrub_min_interval", osd_scrub_min_interval)
+    scrub_object.set_osd_configuration("osd_scrub_max_interval", osd_scrub_max_interval)
+    endTime = datetime.datetime.now() + datetime.timedelta(minutes=60)
+    while datetime.datetime.now() <= endTime:
+        if operation == "scrub":
+            log.debug(f"Running scrub on pg : {pg_id}")
+            if rados_obj.start_check_scrub_complete(pg_id=pg_id):
```

Comment:
Modify the `start_check_scrub_complete` to make the scrub initiation in the method optional

Reply:
Code modified.